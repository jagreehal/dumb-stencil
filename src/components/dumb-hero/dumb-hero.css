:host {
  background: #eee;
  display: block;
  position: relative;
}

.c-hero {
  background: #eee;
  position: relative;
}

.c-hero__img {
  display: block;
  height: auto;
  max-height: 80vh;
  object-fit: cover;
  width: 100%;
}

.c-hero__body {
  bottom: 0;
  left: 0;
  padding: 3rem 2rem;
  position: absolute;
  z-index: 1;
}

.c-hero__title {
  font-size: 3rem;
  margin: 0;
}

.c-hero__description {
  font-size: 1.4rem;
  margin: 0;
}
