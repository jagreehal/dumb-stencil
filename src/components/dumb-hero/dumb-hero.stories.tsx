import '@stencil/core';

import {
  storiesOf
} from '@storybook/html';

const stories = storiesOf('Blocks/Hero', module);

stories.add('Default', () => {


  return (
    `<dumb-hero
    hero-img-src="http://via.placeholder.com/1200x650"
    hero-img-alt="Alt Text"
    hero-title="Hero Title"
    hero-description="This is the hero description"
    >
    </dumb-hero>`
  )
});
