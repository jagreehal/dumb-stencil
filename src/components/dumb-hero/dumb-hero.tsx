import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'dumb-hero',
  styleUrl: 'dumb-hero.css',
  shadow: true
})
export class DumbHero {
  @Prop() heroDescription: string;
  @Prop() heroImgAlt: string;
  @Prop() heroImgSrc: string;
  @Prop() heroTitle: string;

  render() {
    return (
      <div class="c-hero">
        <img class="c-hero__img" src={this.heroImgSrc} alt={this.heroImgAlt} />
        <div class="c-hero__body">
          <h2 class="c-hero__title">{this.heroTitle}</h2>
          <p class="c-hero__description">{this.heroDescription}</p>
        </div>
      </div>
    );
  }
}
