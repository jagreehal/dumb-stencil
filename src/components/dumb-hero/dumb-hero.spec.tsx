import { TestWindow } from '@stencil/core/dist/testing';
import { DumbHero } from './dumb-hero';

describe('dumb-primary-nav', () => {
  let element: HTMLElement;
  let window;

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbHero],
      html: `<dumb-hero
      hero-img-src="http://via.placeholder.com/1200x650"
      hero-img-alt="Alt Text"
      hero-title="Hero Title"
      hero-description="This is the hero description"
      >
      </dumb-hero>`
    });
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
