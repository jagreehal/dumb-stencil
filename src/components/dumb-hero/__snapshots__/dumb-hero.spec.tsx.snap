// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-primary-nav should match snapshot 1`] = `
<dumb-hero
  class="hydrated"
  hero-description="This is the hero description"
  hero-img-alt="Alt Text"
  hero-img-src="http://via.placeholder.com/1200x650"
  hero-title="Hero Title"
>
  
  
      
  <div
    class="c-hero"
  >
    <img
      alt="Alt Text"
      class="c-hero__img"
      src="http://via.placeholder.com/1200x650"
    />
    <div
      class="c-hero__body"
    >
      <h2
        class="c-hero__title"
      >
        Hero Title
      </h2>
      <p
        class="c-hero__description"
      >
        This is the hero description
      </p>
    </div>
  </div>
</dumb-hero>
`;
